pandas/tests/frame/methods/test_replace.py

```diff
@@ -1356,3 +1356,83 @@ def test_replace_replacer_dtype(self, replacer):
         result = df.replace({"a": replacer, "b": replacer})
         expected = pd.DataFrame([replacer])
         tm.assert_frame_equal(result, expected)
+
+    def test_replace_real_dont_change_other_columns(self, any_real_dtype):
+        # GH30512
+        dtype = any_real_dtype
+        value_a = 2
+        value_b = 3
+        df = pd.DataFrame({"A": [value_a], "B": [value_b]}, dtype=dtype)
+        result = df.replace(to_replace={value_a: None}).dtypes.drop("A")
+        expected = df.dtypes.drop("A")
+        tm.assert_series_equal(result, expected)
+
+    def test_replace_string_dont_change_other_columns(self, string_dtype):
+        # GH30512
+        dtype = string_dtype
+        value_a = "a"
+        value_b = "b"
+        df = pd.DataFrame({"A": [value_a], "B": [value_b]}, dtype=dtype)
+        result = df.replace(to_replace={value_a: None}).dtypes.drop("A")
+        expected = df.dtypes.drop("A")
+        tm.assert_series_equal(result, expected)
+
+    def test_replace_datetime64_dont_change_other_columns(self, datetime64_dtype):
```

Comment:
can you parameterize all of these cases into a single test

Reply:
I thought about that, but these test functions take fixtures as arguments (e.g. datetime64_dtype is defined as a fixture in `pandas.conftest.py`). Is it possible to parametrize over fixtures? I tried looking around for a way but couldn't find one

Reply:
> Is it possible to parametrize over fixtures?

I think tests.io.excel.test_readers.engine_and_read_ext might be doing something similar to what you need

Reply:
or tests.arrays.test_integer.all_data might be a better fit

Reply:
@jbrockmendel thanks for the tip, having updated the test - is this what you had in mind?

Reply:
update: what I've done is not correct, it generates nearly 90,000 tests (when I made commit 61b8b2c3b09ad77880dd3adb72c0b79692b93220 there were only 101) in `pandas/tests/frame/methods/test_replace.py`

Reply:
update: what I've done is not correct, it generates nearly 90,000 tests (when I made commit 61b8b2c3b09ad77880dd3adb72c0b79692b93220 there were only 101) in `pandas/tests/frame/methods/test_replace.py`

Reply:
update: what I've done is not correct, it generates nearly 90,000 tests (when I made commit 61b8b2c3b09ad77880dd3adb72c0b79692b93220 there were only 101) in `pandas/tests/frame/methods/test_replace.py`

Reply:
@jbrockmendel @jreback if I write
```
@pytest.fixture(
    params=[
        "any_real_dtype",
        "string_dtype",
        "datetime64_dtype",
        "timedelta64_dtype",
        "bool",
        "complex_dtype",
        "bytes_dtype",
        "tz_aware_fixture",
    ]
)
def replace_value_a_value_b(
    request,
    any_real_dtype,
    string_dtype,
    datetime64_dtype,
    timedelta64_dtype,
    complex_dtype,
    bytes_dtype,
    tz_aware_fixture,
):
    if request.param == "any_real_dtype":
        return any_real_dtype, 2, 3
    elif request.param == "string_dtype":
        return string_dtype, "a", "b"
    elif request.param == "datetime64_dtype":
        return datetime64_dtype, pd.Timestamp("2020-01-24"), pd.Timestamp("2020-01-25")
    elif request.param == "timedelta64_dtype":
        return timedelta64_dtype, pd.Timedelta(3), pd.Timedelta(4)
    elif request.param == "bool":
        return bool, 1, 0
    elif request.param == "complex_dtype":
        return complex_dtype, 1 + 2j, 1 + 3j
    elif request.param == "bytes_dtype":
        return bytes_dtype, b"a", b"b"
    elif request.param == "tz_aware_fixture":
        return (
            None,
            Timestamp("20130102", tz=tz_aware_fixture),
            Timestamp("20130103", tz=tz_aware_fixture),
        )
```
then nearly 90,000 tests are collected.

An alternative would involve importing the types in here (e.g. `ALL_REAL_DTYPES`) and iterating over them but, as I understand, importing from `conftest` is discouraged.

Any other tips on how to do this? It seems parametrisation over fixtures has been proposed in pytest, but it's not there yet - https://docs.pytest.org/en/latest/proposals/parametrize_with_fixtures.html